onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2104,10 +2115,14 @@ def aten_feature_dropout(input: TensorType, p: float, train: bool) -> TensorType
     raise NotImplementedError()
 
 
-def aten_fill(self: TensorType, value: TensorType) -> TensorType:
+@torch_op("aten::fill")
+def aten_fill(self: TTensor, value: TTensor) -> TTensor:
     # fill.Tensor(Tensor self, Tensor value) -> Tensor
 
-    raise NotImplementedError()
+    shape = op.Shape(self)
+    value = op.Cast(value, to=FLOAT.dtype)  # the value might be bool type
```

Comment:
Does it mean all results are FLOAT?

Reply:
from the test case, yes, all FLOAT.

Reply:
I don't see why we need to cast to float here? What happens when the inputs are different types?